tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -243,5 +260,72 @@ private static void WriteRequestHeaders(WebRequest request, IDictionary<string,
             var proxyInstance = obj.DuckAs<ILambdaContext>();
             return proxyInstance?.ClientContext?.Custom;
         }
+
+        private class ConverterContractResolver : DefaultContractResolver
+        {
+            // A custom `ContractResolver` is needed to reduce overhead.
+            // https://www.newtonsoft.com/json/help/html/Performance.htm#JsonConverters
+            protected override JsonContract CreateContract(Type objectType)
+            {
+                // Create a contract with custom settings
+                var contract = base.CreateContract(objectType);
+
+                if (objectType == typeof(MemoryStream))
+                {
+                    contract.Converter = MemoryStreamConverter;
+                }
+                else if (objectType == typeof(DateTime))
+                {
+                    contract.Converter = DateTimeConverter;
+                }
+
+                return contract;
+            }
+        }
+
+        private class MemoryStreamJsonConverter : JsonConverter<MemoryStream>
+        {
+            public override void WriteJson(JsonWriter writer, MemoryStream value, JsonSerializer serializer)
+            {
+                if (value == null)
+                {
+                    writer.WriteNull();
+                    return;
+                }
+
+                try
+                {
+                    var base64String = Convert.ToBase64String(value.ToArray());
+                    writer.WriteValue(base64String);
+                }
+                catch (Exception ex)
+                {
+                    Serverless.Debug($"Failed to serialize MemoryStream with the following error: {ex}");
+                    writer.WriteNull();
+                }
+            }
+
+            public override MemoryStream ReadJson(JsonReader reader, Type objectType, MemoryStream existingValue, bool hasExistingValue, JsonSerializer serializer)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private class DateTimeJsonConverter : JsonConverter<DateTime>
+        {
+            public override void WriteJson(JsonWriter writer, DateTime value, JsonSerializer serializer)
+            {
+                var elapsedTime = value - Epoch;
```

Comment:
Just FYI, there's a built in function `ToUnixTimeSeconds()` on `DateTimeOffset` which does a similar thing (though only returns whole seconds). There's also `ToUnixTimeMilliseconds()`. 

I wonder if the serializer should support `DateTimeOffset` too?

Reply:
I'm so sorry, I merged this before checking this comments.
I will address them in another PR, my bad!
--

Let me try it to see if it returns the same. With milliseconds, I'll have to divide it by thousands just to get the same result!